### ISSUE TITLE
Upon upgrade to v1.17.0-rc.3-k3s.1 helm-install-traefik pod has Errors and cannot get new traefik .tgz

### ISSUE THREAD

**Issue by davidnuzik** [rancher/k3s#1228]
**Version:**
v1.0.1 -> v1.17.0-rc.3-k3s.1 upgrade (issue happens upon upgrade)

**Describe the bug**
When you attempt to upgrade k3s to the v1.17 rc's (rc2 and rc3, rc1 not affected) the helm-install-traefik pod will move to an "Error" status and then "CrashLoopBackoff". Upon investigating the logs we see:

```
+ echo Already installed traefik, upgrading
Already installed traefik, upgrading
+ helm_v2 upgrade traefik https://10.43.0.1:443/static/charts/traefik-1.81.0.tgz
Error: failed to download "https://10.43.0.1:443/static/charts/traefik-1.81.0.tgz" (hint: running `helm repo update` may help)
```

**To Reproduce**
Simply install v1.0.1 or v1.0.0 and upgrade to v1.17.0-rc.3-k3s.1
Watch the helm-install-traefik pod status

**Expected behavior**
Traefik upgrades successfully, pod not in error / crashloopbackoff state

**Actual behavior**
Traefik does not upgrade. The pod has an error. helm complains of a failure to download the necessary tgz.

**Additional context**
If we create a pod we can exec into, and attempt to curl the url to the traefik-1.81.0.tgz it will succeed.

Hussein also tried the following:
```
+ helm_v2 upgrade --debug --set-string kubernetes.ingressEndpoint.useDefaultPublishedService=true --set-string metrics.prometheus.enabled=true --set-string rbac.enabled=true --set-string ssl.enabled=true traefik https://10.43.0.1:443/static/charts/traefik-1.81.0.tgz
[debug] SERVER: "127.0.0.1:44134"
Error: no cached repo found. (try 'helm repo update'). open /root/.helm/repository/cache/stable-index.yaml: no such file or directory
```

**Comment by ShylajaDevadiga** [comment]
Fixed in v1.17.0-rc4-k3s.1

**Comment by davidnuzik** [comment]
Needs PR attached.
